File: masiwasi/app/src/main/java/viewmodels/SharedMascotaViewModel.java
package viewmodels;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.util.ArrayList;
import java.util.List;

import models.Mascota;

public class SharedMascotaViewModel extends ViewModel {

    private final MutableLiveData<List<models.Mascota>> mascotas = new MutableLiveData<>(new ArrayList<>());

    public LiveData<List<models.Mascota>> getMascotas() {
        return mascotas;
    }

    public void addMascota(models.Mascota mascota) {
        List<models.Mascota> list = mascotas.getValue();
        list.add(mascota);
        mascotas.setValue(list);
    }
}

File: movil2/mozita/app/src/main/java/viewmodels/SharedMascotaViewModel.kt
package viewmodels

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import models.Mascota

class SharedMascotaViewModel : ViewModel() {

    private val mascotas: MutableLiveData<MutableList<Mascota>> =
        MutableLiveData(ArrayList())

    fun getMascotas(): LiveData<MutableList<Mascota>> {
        return mascotas
    }

    fun addMascota(mascota: Mascota) {
        val list = mascotas.value
        list?.add(mascota)
        mascotas.value = list
    }
}